javaagent-internal-logging-application/src/main/java/io/opentelemetry/javaagent/logging/application/ApplicationLoggerFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.logging.application;
+
+import io.opentelemetry.javaagent.bootstrap.InternalLogger;
+import io.opentelemetry.javaagent.bootstrap.logging.ApplicationLoggerBridge;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+final class ApplicationLoggerFactory extends ApplicationLoggerBridge
+    implements InternalLogger.Factory {
+
+  private final AtomicBoolean installed = new AtomicBoolean();
+  private volatile InternalLogger.Factory actual = null;
+  private final ConcurrentMap<String, ApplicationLogger> inMemoryLoggers =
+      new ConcurrentHashMap<>();
+
+  private final InMemoryLogStore inMemoryLogStore;
+
+  ApplicationLoggerFactory(InMemoryLogStore inMemoryLogStore) {
+    this.inMemoryLogStore = inMemoryLogStore;
+  }
+
+  @Override
+  protected void install(InternalLogger.Factory applicationLoggerFactory) {
+    // just use the first bridge that gets discovered and ignore the rest
```

Comment:
is it valid/possible to have more than one?

Reply:
Yes - for example, if there are multiple classloaders with separate slf4j copies this method may be invoked multiple times.
I'll add a WARNING log here for when that happens.